packages/iota/src/validation/outputs/common.ts

```diff
@@ -0,0 +1,325 @@
+                    METADATA_FEATURE_TYPE,
+                    TAG_FEATURE_TYPE
+                ]
+        }
+    ],
+    [
+        ALIAS_OUTPUT_TYPE,
+        {
+            minUnlockConditions: 2,
+            unlockConditions:
+                [
+                    STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE,
+                    GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE
+                ],
+            features:
+                [
+                    SENDER_FEATURE_TYPE,
+                    METADATA_FEATURE_TYPE
+                ],
+            immutableFeatures:
+                [
+                    ISSUER_FEATURE_TYPE,
+                    METADATA_FEATURE_TYPE
+                ]
+        }
+    ],
+    [
+        FOUNDRY_OUTPUT_TYPE,
+        {
+            minUnlockConditions: 1,
+            unlockConditions: [IMMUTABLE_ALIAS_UNLOCK_CONDITION_TYPE],
+            features: [METADATA_FEATURE_TYPE],
+            immutableFeatures: [METADATA_FEATURE_TYPE]
+        }
+    ],
+    [
+        NFT_OUTPUT_TYPE,
+        {
+            minUnlockConditions: 1,
+            unlockConditions:
+                [
+                    ADDRESS_UNLOCK_CONDITION_TYPE,
+                    STORAGE_DEPOSIT_RETURN_UNLOCK_CONDITION_TYPE,
+                    TIMELOCK_UNLOCK_CONDITION_TYPE,
+                    EXPIRATION_UNLOCK_CONDITION_TYPE
+                ],
+            features:
+                [
+                    SENDER_FEATURE_TYPE,
+                    METADATA_FEATURE_TYPE,
+                    TAG_FEATURE_TYPE
+                ],
+            immutableFeatures:
+                [
+                    ISSUER_FEATURE_TYPE,
+                    METADATA_FEATURE_TYPE
+                ]
+        }
+    ]
+]);
+
+/**
+ * Unlock Condition type names.
+ */
+const UNLOCK_CONDITION_TYPE_NAMES = [
+    "Address Unlock Condition",
+    "Storage Deposit Return Unlock Condition",
+    "Timelock Unlock Condition",
+    "Expiration Unlock Condition",
+    "State Controller Address Unlock Condition",
+    "Governor Address Unlock Condition",
+    "Immutable Alias Address Unlock Condition"
+];
+
+/**
+ * Unlock Condition type names.
+ */
+const FEATURE_TYPE_NAMES = [
+    "Sender Feature",
+    "Issuer Feature",
+    "Metadata Feature",
+    "Tag Feature"
+];
+
+/**
+ * Map Output type to name.
+ */
+const OUTPUT_TYPE_NAMES = new Map([
+    [BASIC_OUTPUT_TYPE, "Basic"],
+    [ALIAS_OUTPUT_TYPE, "Alias"],
+    [FOUNDRY_OUTPUT_TYPE, "Foundry"],
+    [NFT_OUTPUT_TYPE, "NFT"]
+]);
+
+/**
+ * Validate the output common rules.
+ * @param output The output to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateCommonRules(
+    output: SupportedOutputTypes,
+    protocolInfo: INodeInfoProtocol
+): IValidationResult {
+    const results: IValidationResult[] = [];
+    const outputType = output.type;
+    const outputName = OUTPUT_TYPE_NAMES.get(output.type)!;
```

Comment:
Let's avoid the (!) "trust me typescript, i'm pretty pretty sure this will be there" by handling it at the top.
- Extract outputName without "!"
- Extract validation for output type in `const outputTypeValidations = OUTPUT_TYPE_TO_SUPPORTED_PROP_VALUES.get(output.type);`
- Do a check at the top so **both** must be truthy to continute, if not return early with failed vaildation
```
    if (!(outputName && outputTypeValidations)) {
        return { isValid: false, errors: [`Unsupported output for common rules validation (type: ${output.type})`] };
    }
```
- After this we don't need the "!" anywhere and we can use the **checked** values extracted from the maps